### PR TITLE
Section flow table sorting

### PR DESCRIPTION
**Pull request by epeach**
Enable sorting on the teacher dashboard section table through reactabular library.

(Old/New images have different data in the table)
Old Table:
oldtable

New Table:
sortabletable

New Table Sorted by Name:
sortedbyname

### CODE REVIEW

**Comment by tanyaparker** [comment]
Hey, Erin! I can't really review your code but if you include pictures or cases that you'd like for me to manually test I can sign off on that. :)

**Comment by epeach** [comment]
@tanyaparker I added some photos showing the differences between the tables. Thank you!

**Comment by tanyaparker** [comment]
Losing alternating row coloring is no biggie. I would just prefer all the tables (at the very least on the same page/teacher homepage) to have consistent style. Even better if all of our similar React tables had consistent style. But none of that is a ship blocker.

In the new table before you clicked to sort by name, is there any indication that you can sort the table? Can we have it by default sorted by name so they see the triangle always? (Not sure where the spec is on this if I can see why the decision was made to have a state where there is no triangle.)

**Comment by epeach** [comment]
There are white arrows to the side of all the sortable columns, but we could consider boosting the contrast or choosing a different color for the arrows. Additionally, we have the table sorted by recency ( most recent at the top ) when the teacher arrives on this page, which I think makes sense.

File: apps/test/unit/templates/teacherDashboard/SectionTableButtonCellTest.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { assert } from '../../../util/configuredChai';
+import React from 'react';
+import { shallow } from 'enzyme';
+import {UnconnectedSectionTableButtonCell as SectionTableButtonCell} from "../../../../src/templates/teacherDashboard/SectionTableButtonCell";
+
+describe('buttons column', () => {
+  const section = {id: 1, name: 'sectionA', studentCount: 3, code: 'ABC', courseId: 29, scriptId: 168,
+      grade: '5', providerManaged: false, assignmentName: ['CS Discoveries', 'Unit 1: Problem Solving'],
+      assignmentPaths: ['//localhost-studio.code.org:3000/courses/csd', '//localhost-studio.code.org:3000/s/csd1']};
```
Comment (islemaster) [inline apps/test/unit/templates/teacherDashboard/SectionTableButtonCellTest.jsx]: Go ahead and expand this test section out into one property per line.

**Comment by tanyaparker** [comment]
Oh! I didn't even see those at all. I think the white arrows should be some kind of light gray that's darker than the background color of the column header but lighter than the text in the column header.

And let's keep recency as default. Since they won't be able to sort by that with one of the columns, at least they have that option on first page load.

**Comment by epeach** [comment]
@tanyaparker updated the arrows colors

arrows

**Comment by tanyaparker** [comment]
Light gray looks great!

Deliberate we don't sort by course? I guess not particularly useful, but it feels like we should just blanket make all columns sortable for consistency.

**Comment by Erin007** [comment]
Is there planned work to make the student sections table consistent in styling with this table? Also, just wondering, is there RTL support planned for these tables too? There was styling for this at one point, but I don't know if that's at all a priority.

**Comment by Erin007** [comment]
Is the dark down arrow next to sections indicating that it's the "active" column that is being sorted?

File: apps/src/templates/teacherDashboard/SectionTable.jsx [hunk 2]
```diff
@@ -1,33 +1,129 @@
 import React, { Component, PropTypes } from 'react';
 import { connect } from 'react-redux';
 import color from "@cdo/apps/util/color";
-import SectionRow from './SectionRow';
+import {Table, sort} from 'reactabular';
 import i18n from '@cdo/locale';
 import { styles as tableStyles } from '@cdo/apps/templates/studioHomepages/SectionsTable';
```
Comment (Erin007) [inline apps/src/templates/teacherDashboard/SectionTable.jsx:6]: is this shared styling still applicable? If not, we can nix the import and I think change the export and comment referencing it in `SectionsTable`.

**Comment by tanyaparker** [comment]
Good questions, @Erin007. I think this PR must include continued support for RTL. Making the student's section table (and other tables) have the new style with same colored rows is on Werketing backlog.

**Comment by epeach** [comment]
I'll do my best to answer these:
Not sorting by course is deliberate. It's not impossible, but would involve some additional work before it could be set to go.
The dark arrow does indicate the column and direction of the current sorting.

**Comment by epeach** [comment]
@tanyaparker and @Erin007 Please check out the RTL screenshot and let me know if anything needs to be updated. Cheers!

rtl

**Comment by Erin007** [comment]
Looks good to me. Thanks, Erin!

**Comment by tanyaparker** [comment]
Gotcha. If sorting by course isn't trivial, forget it.

**Pull request merged**